### PR TITLE
Addresses #10: Created new hooks to record

### PR DESCRIPTION
**Pull request by amol**
New Request Hook has been added to create TimedClient and new Response
Hook has been created to log the time used by individual requests.

### CODE REVIEW

File: jumpgate/common/hooks/sl/timedclient.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from SoftLayer import TimedClient
+from oslo.config import cfg
+from jumpgate.common.hooks import request_hook
+from jumpgate.common.sl.auth import get_auth
+import time
```
Comment (bodenr) [inline jumpgate/common/hooks/sl/timedclient.py]: style -- pep8 prefers std imports atop lib specific.... see http://legacy.python.org/dev/peps/pep-0008/#imports

File: jumpgate/common/hooks/sl/timelog.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import logging
+
+from jumpgate.common.hooks import response_hook
+import time
```
Comment (bodenr) [inline jumpgate/common/hooks/sl/timelog.py]: style -- same comment as above on pep8 import guidelines

**Comment by bodenr** [comment]
random thought -- we already have support for a client request hook and logging response hook... rather than intro another set of those hooks for the timed version, is it more user friendly to just reuse them and add a flag to indicate using the timed version?

**Comment by amol** [comment]
I also thought about it, but not sure how to pass parameter to hooks. We can use config to determine if we want logger to record the time. 

``` conf
[softlayer]
log_timing=True
```

@sudorandom can you confirm if this needs to be done?

File: jumpgate/common/hooks/sl/timelog.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    end_time = time.time()
+    start_time = req.env.get('sl_timehook_start_time', None)
+    if not start_time:
+        LOG.error(
+            "timelog can only be used along with timedclient request hook")
+        return
+    timed_client = req.env['sl_client']
+    overall = end_time - start_time
+    sl_total = 0
+    for timed_call in timed_client.get_last_calls():
```
Comment (sudorandom) [inline jumpgate/common/hooks/sl/timelog.py]: In python you can expand a tuple like this:

``` python
for call, time_stamp, duration in timed_client.get_last_calls():
```

**Comment by amol** [comment]
Can this be merged?

**Comment by sudorandom** [comment]
This LGTM, sorry for the delay.

**Comment by sudorandom** [comment]
As for the option suggestion that Boden has, we can address that in another pull request if needed. I would prefer that it be an option, but you'd still have to configure the after hook. I'm curious what @beittenc 's opinion on this is.

**Pull request merged**